packages/sqrl-text-functions/src/InProcessPatternService.ts

```diff
@@ -15,11 +15,11 @@ function escapeRegExp(str: string): string {
   return str.replace(matchOperatorsRe, "\\$&");
 }
 
-interface GenericRegExp {
-  lastIndex: number;
-  exec(content: string): string[]
-}
-export type CreateRegExp = (source, flags) => GenericRegExp;
+type GenericRegExp = Pick<RegExp, "lastIndex" | "exec">;
+
+export type CreateRegExp = (
+  ...args: Parameters<typeof RegExp>
+) => GenericRegExp;
```

Comment:
@qix I noticed that `source` and `flags` are implicitly set to `any`. Not sure what your feelings are around `noImplicitAny` and stricter types in general but I think we can probably crank the strictness up a little without making the dev experience a nightmare.

Reply:
@meyer yeah I am good with stricter types, the implicit any was just a relic from the super old codebase.